src/include/time_inline.h

```diff
@@ -169,9 +169,8 @@ __wt_op_timer_start(WT_SESSION_IMPL *session)
 {
     uint64_t timeout_us;
 
-    /* Timer can be configured per-transaction, and defaults to per-connection. */
-    if (session->txn == NULL || (timeout_us = session->txn->operation_timeout_us) == 0)
-        timeout_us = S2C(session)->operation_timeout_us;
+    /* Timer is configured per-transaction if there is a value. */
+    timeout_us = (session->txn == NULL) ? 0 : session->txn->operation_timeout_us;
     if (timeout_us == 0)
         session->operation_start_us = session->operation_timeout_us = 0;
     else {
```

Comment:
This function could be simplified a bit as most of the variables depend on eachother. Two possible simplifications are:
No local timeout var:
```
    if (session->txn == NULL || session->txn->operation_timeout_us == 0) {
        session->operation_start_us = 0;
    } else {
        session->operation_timeout_us = session->txn->operation_timeout_us;
        session->operation_start_us = __wt_clock(session);
    }
```
Just ternaries:
```
    uint64_t timeout_us;

    timeout_us = (session->txn == NULL) ? 0 : session->txn->operation_timeout_us;

    session->operation_start_us = timeout_us == 0 ? 0 : __wt_clock(session);
    session->operation_timeout_us = timeout_us;
```
I think #2 is a bit more readable but I'll leave it up to you to decide.

Reply:
Fixed in [4de96ae](https://github.com/wiredtiger/wiredtiger/pull/9357/commits/4de96ae50ca00ba0bde584aabe445769f4f6ac27)